front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineTemplateController.java

```diff
@@ -69,6 +80,41 @@ PipelineTemplate update(@PathVariable String id, @RequestBody PipelineTemplate p
     return pipelineTemplate;
   }
 
+  @RequestMapping(value = "{id}", method = RequestMethod.DELETE)
+  void delete(@PathVariable String id) {
+    checkForDependentConfigs(id);
+    getPipelineTemplateDAO().delete(id);
+  }
+
+  @VisibleForTesting
+  void checkForDependentConfigs(String templateId) {
```

Comment:
You'd also want to check for dependent templates, since templates can inherit from another.